oap-application/oap-application/src/main/java/oap/application/supervision/Supervisor.java

```diff
@@ -38,20 +49,52 @@ public class Supervisor {
 
     private boolean stopped = false;
 
+    private static void runAndDetectTimeout( String name, ShutdownConfiguration shutdownConfiguration, Runnable func ) {
+        if( shutdownConfiguration.timeoutMs > 0 ) {
+            long start = DateTimeUtils.currentTimeMillis();
+            Future<?> future = shutdownConfiguration.submit( func );
+            try {
+                future.get( shutdownConfiguration.timeoutMs, TimeUnit.MILLISECONDS );
+            } catch( InterruptedException e ) {
+                log.trace( e.getMessage() );
+            } catch( ExecutionException e ) {
+                throw Throwables.propagate( e );
+            } catch( TimeoutException e ) {
+                log.warn( "APP_TIMEOUT_START service {} after {}", name, Dates.durationToString( shutdownConfiguration.timeoutMs ) );
+
+                try {
+                    if( !shutdownConfiguration.forceAsyncAfterTimeout ) {
```

Comment:
So it is tested in `testLifecycle()` @Test, right?
Is there a conf with `forceAsyncAfterTimeout` true and false?

Reply:
APPLICATION_FORCE_ASYNC_AFTER_TIMEOUT = true if value in [on, ON, yes, YES, true, TRUE, 1]